### PR TITLE
Formatting, licensing, ang Git hooks

### PR DESCRIPTION
**Pull request by robertodr**
- [x] Propose a `.clang-format` file. @MRChemSoft/mrchemdev I have formatted `src/scf_solver` Please look at the diff to check whether you like it or not. If not, let me know what you'd like changed. I ran the formatting with: `cd src/scf_solver; git ls-files -- '*.cpp' '*.h' | xargs clang-format -i`
- [x] Add (optional) Git pre-commit hooks to check that licensing headers and formatting are correct.

To add the license header on top of the source files:
```
./.githooks/pre-commit-license-maintainer
```
I am not doing since it would touch all files and I think it's better @stigrj does something like that.

### CODE REVIEW

File: src/scf_solver/Accelerator.cpp [hunk 1]
```diff
@@ -41,12 +41,8 @@ void Accelerator::clear() {
         orbital::free(this->dOrbitals[0]);
         this->dOrbitals.pop_front();
     }
-    while (this->fock.size() > 0) {
-        this->fock.pop_front();
-    }
-    while (this->dFock.size() > 0) {
-        this->dFock.pop_front();
-    }
+    while (this->fock.size() > 0) { this->fock.pop_front(); }
```
Comment (stigrj) [inline src/scf_solver/Accelerator.cpp]: For one-liners I like to skip the `{` `}`, but it's probably not for the formatter to remove those?

**Comment by stigrj** [comment]
Looks very good to me. What's the condition for line break? Some of the argument lists look awfully long

File: src/scf_solver/Accelerator.cpp (hunk 1 above)
Comment (robertodr) [reply src/scf_solver/Accelerator.cpp]: Let me see if I can convince the formatter to not put braces for single-line constructs.

**Comment by robertodr** [comment]
The breaks are computed based on the column number. This is now set to 120, trying with 85 didn't help that much. And less than 85 seems silly.

**Comment by stigrj** [comment]
Ok, just keep 120 then

File: src/scf_solver/Accelerator.cpp (hunk 1 above)
Comment (robertodr) [reply src/scf_solver/Accelerator.cpp]: Can't manage to convince to remove braces...

**Comment by robertodr** [comment]
I've unmade some of the argument list changes by hand. These are now protected with `// clang-format off` and `// clang-format on`

**Comment by stigrj** [comment]
Hmm, I think I prefer the long arguments over the extra comments :stuck_out_tongue:

File: src/scf_solver/Accelerator.cpp (hunk 1 above)
Comment (stigrj) [reply src/scf_solver/Accelerator.cpp]: That's fine, it doesn't seem to enforce them either so we just have to avoid writing them :slightly_smiling_face:

**Comment by stigrj** [comment]
So what would you like me to do next?

**Comment by robertodr** [comment]
This one can be merged. You can then apply license header repo-wide and that shouldn't cause too many conflicts. Formatting might generate a lot of conflicts for the rest of the team, so that might have to be discussed more carefully.

**Pull request merged**